components/renku_data_services/session/orm.py

```diff
@@ -223,6 +244,23 @@ def dump(self) -> models.BuildParameters:
         )
 
 
+class BuildPlatformORM(BaseORM):
+    """The build platforms referenced by build parameters."""
+
+    __tablename__ = "build_platforms"
+
+    id: Mapped[int] = mapped_column(Integer, Identity(always=True), primary_key=True, init=False)
+
+    platform: Mapped[str] = mapped_column("platform", String(99))
+
+    build_parameters_id: Mapped[ULID] = mapped_column(
+        "build_parameters_id",
+        ForeignKey(BuildParametersORM.id, ondelete="CASCADE", name="build_platform_build_parameters_id_fk"),
+        init=False,
+    )
+    build_parameters: Mapped[BuildParametersORM] = relationship(back_populates="platforms", lazy="select", default=None)
```

Comment:
issue: we need a uniqueness constraint so that for a specific build_parameters_id we have only unique platforms.

Reply:
> issue: we need a uniqueness constraint so that for a specific build_parameters_id we have only unique platforms.

No. The uniqueness only comes from us not having a ready-made build strategy for multi-platform builds. Multi-platform builds are semantically correct and could be supported very soon. This means that this is NOT a constraint we should put in the database, but only at the input validation layer.